src/leds.c

```diff
@@ -0,0 +1,73 @@
+#include "leds.h"
+
+#include "ch.h"
+#include "hal.h"
+#include "board.h"
+
+enum LedState actLed;
+enum LedState statLed;
+enum LedState errLed;
+uint32_t Counter = 0;
+
+void init_leds(void)
+{
+	palClearPad(GPIOB, GPIOB_LED_ERR);
+	palClearPad(GPIOB, GPIOB_LED_STAT);
+	palClearPad(GPIOB, GPIOB_LED_ACT);
+
+	actLed = led_off;
+	statLed = led_off;
+	errLed = led_off;
+	Counter = 0;
+}
+
+void set_led(enum LedSelect led, enum LedState state)
+{
+	switch(led)
+	{
+		case led_act:
+			actLed = state;
+			break;
+		case led_stat:
+			statLed = state;
+			break;
+		case led_err:
+			errLed = state;
+			break;
+	}
+
+	if(state == led_off)
+		palClearPad(GPIOB, led);
+	else if(state == led_on)
+		palSetPad(GPIOB, led);
+
+}
+
+void toggle_led(enum LedSelect led)
+{
+	palTogglePad(GPIOB, led);
+}
+
+void update_leds(void)
+{
+	if(actLed == led_blink_250){ toggle_led(led_act); }
```

Comment:
the linux checkpatch linter will come after you :+) single line is enough without braces and should be placed on the next line